skippkg-finder.py

```diff
@@ -0,0 +1,333 @@
+                if orig_name in pkglist and pkglist[orig_name]['Project'] == orig_project:
+                    if re.match(r'[0-9]+$', pkglist[orig_name]['Package'].split('.')[-1]):
+                        old_incident_number = int(pkglist[orig_name]['Package'].split('.')[-1])
+                        if incident_number > old_incident_number:
+                            pkglist[orig_name]['Package'] = pkgname
+                    else:
+                        pkglist[orig_name]['Package'] = pkgname
+            else:
+                pkglist[pkgname] = {'Project': orig_project, 'Package': pkgname}
+
+        if by_project:
+            for pkg in pkglist.keys():
+                if pkglist[pkg]['Project'].startswith('SUSE:') and self.is_sle_specific(pkg):
+                    continue
+                if pkglist[pkg]['Project'] not in packageinfo:
+                    packageinfo[pkglist[pkg]['Project']] = []
+                if pkglist[pkg]['Package'] not in packageinfo[pkglist[pkg]['Project']]:
+                    packageinfo[pkglist[pkg]['Project']].append(pkglist[pkg]['Package'])
+            return packageinfo
+
+        return pkglist
+
+    def get_project_binary_list(self, project, repository, arch, package_binaries={}):
+        """
+        Returns binarylist of a project
+        """
+
+        if project.startswith('SUSE:'):
+            repository = 'pool'
+
+        path = ['build', project, repository, arch]
+        url = makeurl(self.apiurl, path, {'view': 'binaryversions'})
+        root = ET.parse(http_GET(url)).getroot()
+
+        for binary_list in root:
+            package = binary_list.get('package')
+            package = package.split(':', 1)[0]
+            index = project + "_" + package
+
+            if index not in package_binaries:
+                package_binaries[index] = []
+            for binary in binary_list:
+                filename = binary.get('name')
+                result = re.match(RPM_REGEX, filename)
+                if not result:
+                    continue
+
+                if result.group('arch') == 'src' or result.group('arch') == 'nosrc':
+                    continue
+                if result.group('name').endswith('-debuginfo') or result.group('name').endswith('-debuginfo-32bit'):
+                    continue
+                if result.group('name').endswith('-debugsource'):
+                    continue
+
+                if result.group('name') not in package_binaries[index]:
+                    package_binaries[index].append(result.group('name'))
+
+        return package_binaries
+
+    def item_exists(self, project, package=None):
+        """
+        Return true if the given project or package exists
+        """
+
+        if package:
+            url = makeurl(self.apiurl, ['source', project, package, '_meta'])
+        else:
+            url = makeurl(self.apiurl, ['source', project, '_meta'])
+        try:
+            http_GET(url)
+        except HTTPError:
```

Comment:
this is error prone as it does not need to be 404 but e.g. also temporary 500. We also have similar method https://github.com/openSUSE/openSUSE-release-tools/pull/2623/files#diff-22a4ac30776b144d4e04816a4e9ccfc2f8999d985f675e160c7a21bf88d2a775R87